instagram-clone/src/authentication/withAuthenticate.js

```diff
@@ -0,0 +1,34 @@
+import React from 'react'
+
+const withAuthenticate = PostsPage => LoginPage =>
+
+ class extends React.Component {
+
+    constructor() {
+        super()
+
+         this.state = {
+             isLoggedin: false
+         }
+    }
+
+ componentDidMount(){
+        if(localStorage.getItem("user")){
+            this.setState({isLoggedin: true})
+        }else{
+            this.setState({isLoggedin: false})
+        }
+    }
+
+render() {
```

Comment:
Nice work here. We could clean it up with ternary statements. This is a personal preference for me, explicit `if/else` statements can be easier to read for some people.
```
    componentDidMount() {
            this.setState({ 
                isLoggedIn: localStorage.getItem('user') ? true : false 
            })
    }

       render() {
              return this.state.isLoggedIn ? <PostsPage /> : <LoginPage />
      }
```